### PR TITLE
Ensure room is actually in space hierarchy when resolving its latest version

### PR DESCRIPTION
**Pull request by Johennes**
This fixes one of the issues from vector-im/element-web#24329

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] <del>Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))</del>

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Ensure room is actually in space hierarchy when resolving its latest version ([\#10010](https://github.com/matrix-org/matrix-react-sdk/pull/10010)).

### CODE REVIEW

**Comment by Johennes** [comment]
Remaining static analysis errors are in a file I didn't touch. 🤷‍♂️

**Review by florianduros — CHANGES_REQUESTED** [review]
Seems good! If unit tests can be added to cover theses changes, it will be great :)

Thanks!

**Comment by Johennes** [comment]
You passed the attentive reviewer test. 😅

Have added tests, too, now. 🙂

**Comment by Johennes** [comment]
Hm, interestingly Sonarcloud _only now_ complains about the coverage being slightly below 80%. 🤔

**Comment by Johennes** [comment]
@florianduros coverage is above 80% now. Would appreciate feedback on the quality of tests as I'm not sure if these are very great.

File: test/components/structures/SpaceHierarchy-test.tsx [hunk 1]
```diff
@@ -67,4 +79,137 @@ describe("SpaceHierarchy", () => {
...
         });
     });
+
+    describe("toLocalRoom", () => {
+        let client: MatrixClient;
+        let roomV1: Room;
+        let roomV2: Room;
+        let roomV3: Room;
+
+        beforeEach(() => {
```
Comment (florianduros) [inline test/components/structures/SpaceHierarchy-test.tsx:89]: If we are not mutating the mock created in the `beforeEach`, we can created them in the `describe` scope.

File: test/components/structures/SpaceHierarchy-test.tsx [hunk 2]
```diff
@@ -67,4 +79,137 @@ describe("SpaceHierarchy", () => {
...
+        let root: Room;
+        let room1: Room;
+        let room2: Room;
+
+        let hierarchyRoot: IHierarchyRoom;
+        let hierarchyRoom1: IHierarchyRoom;
+        let hierarchyRoom2: IHierarchyRoom;
+
+        let roomHierarchy: RoomHierarchy;
+
```
Comment (florianduros) [inline test/components/structures/SpaceHierarchy-test.tsx:153]: If we are not mutating the mock created in the beforeEach, we can created them in the describe scope.

**Review by florianduros — APPROVED** [review]
Looks good! Imo, a small improvement can be made in the test :)

**Pull request merged**